package.json

```diff
@@ -87,11 +95,13 @@
   "types": "./tsdef/index.d.ts",
   "scripts": {
     "lint:js": "eslint ./lib ./test/*.js ./docker/**/*.js ./test/framework/*.js ./test/lib/*.js ./test/integration/** ./test/unit/** ",
-    "lint:ts": "eslint ./tsdef/*.ts",
+    "lint:ts": "eslint ./tsdef/*.ts ./lib/**/*.ts",
     "lint": "npm-run-all lint:js lint:ts",
-    "test:unit": "mocha ./test/unit/index.js",
+    "test:unit": "npm-run-all test:unit:js test:unit:ts",
+    "test:unit:js": "nyc --report-dir=./coverage/js --extension=.js --include=lib/**/* --reporter=html --reporter=lcov --reporter=text mocha ./test/unit/index.js",
+    "test:unit:ts": "nyc --report-dir=./coverage/ts --extension=.ts --include=lib/**/* --reporter=html --reporter=lcov --reporter=text mocha -r ts-node/register ./test/unit/index.ts",
     "test:integration:adapter": "node ./scripts/karma.js karma/integration.adapter.conf.js",
-    "test:integration": "node ./scripts/karma.js karma/integration.conf.js",
+    "test:integration": "npm run build:es5 && node ./scripts/karma.js karma/integration.conf.js",
```

Comment:
did you look into [karma-typescript](https://www.npmjs.com/package/karma-typescript) to run our integration tests?

Reply:
Yes. See PR description that describes the issue of using karma-typescript in our codebase.